[Bug]: Cannot connect to remote postgres in docker

### Describe the bug

Does memmachine support remote postgres db, like supabase? I was trying to run ./memmachine-compose.sh locally, but it failed:
```
 ✔ Network memmachine-network           Created                 0.0s 
 ✔ Volume "memmachine_memmachine_logs"  Created                 0.0s 
 ✔ Volume "memmachine_postgres_data"    Created                 0.0s 
 ✔ Volume "memmachine_neo4j_data"       Created                 0.0s 
 ✔ Volume "memmachine_neo4j_logs"       Created                 0.0s 
 ✔ Volume "memmachine_neo4j_import"     Created                 0.0s 
 ✔ Volume "memmachine_neo4j_plugins"    Created                 0.0s 
 ✔ Container memmachine-neo4j           Healthy                32.8s 
 ✔ Container memmachine-postgres        Healthy                10.8s 
 ✔ Container memmachine-app             Started                32.9s 
[SUCCESS] Services started successfully!
[INFO] Waiting for services to be healthy...
NAME                  IMAGE                    COMMAND                  SERVICE             CREATED             STATUS                                     PORTS
memmachine-app        memmachine/memmachine    "sh -c 'memmachine-s…"   memmachine          33 seconds ago      Up Less than a second (health: starting)   0.0.0.0:8080->8080/tcp
memmachine-neo4j      neo4j:5.23-community     "tini -g -- /startup…"   neo4j               33 seconds ago      Up 32 seconds (healthy)                    0.0.0.0:7473-7474->7473-7474/tcp, 0.0.0.0:7687->7687/tcp
memmachine-postgres   pgvector/pgvector:pg16   "docker-entrypoint.s…"   postgres            33 seconds ago      Up 32 seconds (healthy)                    0.0.0.0:5432->5432/tcp
[INFO] Checking service health...
[INFO] Waiting for PostgreSQL to be ready...
./memmachine-compose.sh: line 160: timeout: command not found
[ERROR] PostgreSQL failed to become ready in 120 seconds. Check container logs and configuration.
```

logs in docker memmachine-postgres:
```
2025-09-18 23:58:51 The files belonging to this database system will be owned by user "postgres".
2025-09-18 23:58:51 This user must also own the server process.
2025-09-18 23:58:51 
2025-09-18 23:58:51 The database cluster will be initialized with this locale configuration:
2025-09-18 23:58:51   provider:    libc
2025-09-18 23:58:51   LC_COLLATE:  C
2025-09-18 23:58:51   LC_CTYPE:    C
2025-09-18 23:58:51   LC_MESSAGES: en_US.utf8
2025-09-18 23:58:51   LC_MONETARY: en_US.utf8
2025-09-18 23:58:51   LC_NUMERIC:  en_US.utf8
2025-09-18 23:58:51   LC_TIME:     en_US.utf8
2025-09-18 23:58:51 The default text search configuration will be set to "english".
2025-09-18 23:58:51 
2025-09-18 23:58:51 Data page checksums are disabled.
2025-09-18 23:58:51 
2025-09-18 23:58:51 fixing permissions on existing directory /var/lib/postgresql/data ... ok
2025-09-18 23:58:51 creating subdirectories ... ok
2025-09-18 23:58:51 selecting dynamic shared memory implementation ... posix
2025-09-18 23:58:51 selecting default max_connections ... 100
2025-09-18 23:58:51 selecting default shared_buffers ... 128MB
2025-09-18 23:58:51 selecting default time zone ... Etc/UTC
2025-09-18 23:58:51 creating configuration files ... ok
2025-09-18 23:58:51 running bootstrap script ... ok
2025-09-18 23:58:51 performing post-bootstrap initialization ... ok
2025-09-18 23:58:51 syncing data to disk ... ok
2025-09-18 23:58:51 
2025-09-18 23:58:51 
2025-09-18 23:58:51 Success. You can now start the database server using:
2025-09-18 23:58:51 initdb: warning: enabling "trust" authentication for local connections
2025-09-18 23:58:51 initdb: hint: You can change this by editing pg_hba.conf or using the option -A, or --auth-local and --auth-host, the next time you run initdb.
2025-09-18 23:58:51 2025-09-19 06:58:51.933 UTC [1] LOG:  starting PostgreSQL 16.10 (Debian 16.10-1.pgdg12+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14+deb12u1) 12.2.0, 64-bit
2025-09-18 23:58:51 2025-09-19 06:58:51.933 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2025-09-18 23:58:51 2025-09-19 06:58:51.933 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2025-09-18 23:58:51 2025-09-19 06:58:51.934 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2025-09-18 23:58:51 2025-09-19 06:58:51.938 UTC [62] LOG:  database system was shut down at 2025-09-19 06:58:51 UTC
2025-09-18 23:58:51 2025-09-19 06:58:51.943 UTC [1] LOG:  database system is ready to accept connections
2025-09-18 23:58:51 
2025-09-18 23:58:51     pg_ctl -D /var/lib/postgresql/data -l logfile start
2025-09-18 23:58:51 
2025-09-18 23:58:51 waiting for server to start....2025-09-19 06:58:51.703 UTC [48] LOG:  starting PostgreSQL 16.10 (Debian 16.10-1.pgdg12+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14+deb12u1) 12.2.0, 64-bit
2025-09-18 23:58:51 2025-09-19 06:58:51.703 UTC [48] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2025-09-18 23:58:51 2025-09-19 06:58:51.707 UTC [51] LOG:  database system was shut down at 2025-09-19 06:58:51 UTC
2025-09-18 23:58:51 2025-09-19 06:58:51.711 UTC [48] LOG:  database system is ready to accept connections
2025-09-18 23:58:51  done
2025-09-18 23:58:51 server started
2025-09-18 23:58:51 
2025-09-18 23:58:51 /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
2025-09-18 23:58:51 
2025-09-18 23:58:51 waiting for server to shut down...2025-09-19 06:58:51.813 UTC [48] LOG:  received fast shutdown request
2025-09-18 23:58:51 .2025-09-19 06:58:51.814 UTC [48] LOG:  aborting any active transactions
2025-09-18 23:58:51 2025-09-19 06:58:51.817 UTC [48] LOG:  background worker "logical replication launcher" (PID 54) exited with exit code 1
2025-09-18 23:58:51 2025-09-19 06:58:51.818 UTC [49] LOG:  shutting down
2025-09-18 23:58:51 2025-09-19 06:58:51.819 UTC [49] LOG:  checkpoint starting: shutdown immediate
2025-09-18 23:58:51 2025-09-19 06:58:51.822 UTC [49] LOG:  checkpoint complete: wrote 3 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.002 s, sync=0.001 s, total=0.004 s; sync files=2, longest=0.001 s, average=0.001 s; distance=0 kB, estimate=0 kB; lsn=0/14EA210, redo lsn=0/14EA210
2025-09-18 23:58:51 2025-09-19 06:58:51.826 UTC [48] LOG:  database system is shut down
2025-09-18 23:58:51  done
2025-09-18 23:58:51 server stopped
2025-09-18 23:58:51 
2025-09-18 23:58:51 PostgreSQL init process complete; ready for start up.
2025-09-18 23:58:51 
2025-09-19 00:03:52 2025-09-19 07:03:52.012 UTC [60] LOG:  checkpoint starting: time
2025-09-19 00:03:56 2025-09-19 07:03:56.271 UTC [60] LOG:  checkpoint complete: wrote 44 buffers (0.3%); 0 WAL file(s) added, 0 removed, 0 recycled; write=4.231 s, sync=0.011 s, total=4.261 s; sync files=11, longest=0.007 s, average=0.001 s; distance=260 kB, estimate=260 kB; lsn=0/152B630, redo lsn=0/152B5F8
```

ogs in docker memmachine-app:
```
2025-09-18 23:59:24 Syncing schema to {'host': 'db.tloddrldpecozlxxpvpm.supabase.co', 'port': '5432', 'user': 'postgres', 'password': '****', 'database': 'postgres'}
2025-09-18 23:59:24 Traceback (most recent call last):
2025-09-18 23:59:24   File "/app/.venv/bin/memmachine-sync-profile-schema", line 10, in <module>
2025-09-18 23:59:24     sys.exit(main())
2025-09-18 23:59:24              ^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/memmachine/profile_memory/storage/syncschema.py", line 117, in main
2025-09-18 23:59:24     asyncio.run(main_async(args))
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/runners.py", line 195, in run
2025-09-18 23:59:24     return runner.run(main)
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/runners.py", line 118, in run
2025-09-18 23:59:24     return self._loop.run_until_complete(task)
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
2025-09-18 23:59:24     return future.result()
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/memmachine/profile_memory/storage/syncschema.py", line 125, in main_async
2025-09-18 23:59:24     await sync_to(
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/memmachine/profile_memory/storage/syncschema.py", line 73, in sync_to
2025-09-18 23:59:24     connection = await asyncpg.connect(**d)
2025-09-18 23:59:24                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connection.py", line 2421, in connect
2025-09-18 23:59:24     return await connect_utils._connect(
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connect_utils.py", line 1075, in _connect
2025-09-18 23:59:24     raise last_error or exceptions.TargetServerAttributeNotMatched(
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connect_utils.py", line 1049, in _connect
2025-09-18 23:59:24     conn = await _connect_addr(
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connect_utils.py", line 886, in _connect_addr
2025-09-18 23:59:24     return await __connect_addr(params, True, *args)
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connect_utils.py", line 931, in __connect_addr
2025-09-18 23:59:24     tr, pr = await connector
2025-09-18 23:59:24              ^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connect_utils.py", line 802, in _create_ssl_connection
2025-09-18 23:59:24     tr, pr = await loop.create_connection(
2025-09-18 23:59:24              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1132, in create_connection
2025-09-18 23:59:24     raise exceptions[0]
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1107, in create_connection
2025-09-18 23:59:24     sock = await self._connect_sock(
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1010, in _connect_sock
2025-09-18 23:59:24     await self.sock_connect(sock, address)
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/selector_events.py", line 651, in sock_connect
2025-09-18 23:59:24     return await fut
2025-09-18 23:59:24            ^^^^^^^^^
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/selector_events.py", line 659, in _sock_connect
2025-09-18 23:59:24     sock.connect(address)
2025-09-18 23:59:24 OSError: [Errno 99] Cannot assign requested address
2025-09-18 23:59:24 Traceback (most recent call last):
2025-09-18 23:59:24   File "/app/.venv/bin/memmachine-sync-profile-schema", line 10, in <module>
2025-09-18 23:59:24     sys.exit(main())
2025-09-18 23:59:24              ^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/memmachine/profile_memory/storage/syncschema.py", line 117, in main
2025-09-18 23:59:24     asyncio.run(main_async(args))
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/runners.py", line 195, in run
2025-09-18 23:59:24     return runner.run(main)
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/runners.py", line 118, in run
2025-09-18 23:59:24     return self._loop.run_until_complete(task)
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
2025-09-18 23:59:24     return future.result()
2025-09-18 23:59:24 Syncing schema to {'host': 'db.tloddrldpecozlxxpvpm.supabase.co', 'port': '5432', 'user': 'postgres', 'password': '****', 'database': 'postgres'}
2025-09-18 23:59:25 Syncing schema to {'host': 'db.tloddrldpecozlxxpvpm.supabase.co', 'port': '5432', 'user': 'postgres', 'password': '****', 'database': 'postgres'}
2025-09-18 23:59:25 Syncing schema to {'host': 'db.tloddrldpecozlxxpvpm.supabase.co', 'port': '5432', 'user': 'postgres', 'password': '****', 'database': 'postgres'}
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/memmachine/profile_memory/storage/syncschema.py", line 125, in main_async
2025-09-18 23:59:24     await sync_to(
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/memmachine/profile_memory/storage/syncschema.py", line 73, in sync_to
2025-09-18 23:59:24     connection = await asyncpg.connect(**d)
2025-09-18 23:59:24                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connection.py", line 2421, in connect
2025-09-18 23:59:24     return await connect_utils._connect(
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connect_utils.py", line 1075, in _connect
2025-09-18 23:59:24     raise last_error or exceptions.TargetServerAttributeNotMatched(
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connect_utils.py", line 1049, in _connect
2025-09-18 23:59:24     conn = await _connect_addr(
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connect_utils.py", line 886, in _connect_addr
2025-09-18 23:59:24     return await __connect_addr(params, True, *args)
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connect_utils.py", line 931, in __connect_addr
2025-09-18 23:59:24     tr, pr = await connector
2025-09-18 23:59:24              ^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/app/.venv/lib/python3.12/site-packages/asyncpg/connect_utils.py", line 802, in _create_ssl_connection
2025-09-18 23:59:24     tr, pr = await loop.create_connection(
2025-09-18 23:59:24              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1132, in create_connection
2025-09-18 23:59:24     raise exceptions[0]
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1107, in create_connection
2025-09-18 23:59:24     sock = await self._connect_sock(
2025-09-18 23:59:24            ^^^^^^^^^^^^^^^^^^^^^^^^^
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1010, in _connect_sock
2025-09-18 23:59:24     await self.sock_connect(sock, address)
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/selector_events.py", line 651, in sock_connect
2025-09-18 23:59:24     return await fut
2025-09-18 23:59:24            ^^^^^^^^^
2025-09-18 23:59:24   File "/usr/local/lib/python3.12/asyncio/selector_events.py", line 659, in _sock_connect
2025-09-18 23:59:24     sock.connect(address)
2025-09-18 23:59:24 OSError: [Errno 99] Cannot assign requested address
```

### Steps to reproduce

Use a remote postgres connection (e.g., supabase) and run `./memmachine-compose.sh`

### Expected behavior

It should be able to connect to remote postgres

### Environment

MacOS, Python 3.12

### Additional context

_No response_

--- a/src/memmachine/episodic_memory/long_term_memory/long_term_memory.py
+++ b/src/memmachine/episodic_memory/long_term_memory/long_term_memory.py
@@ -58,11 +58,14 @@ class LongTermMemory:
         if not isinstance(neo4j_host, str):
             raise TypeError("Neo4j host must be provided as a string")
 
-        neo4j_port = vector_graph_store_config.get("port")
-        if not isinstance(neo4j_port, int):
-            raise TypeError("Neo4j port must be provided as an integer")
+        if "neo4j+s://" in neo4j_host:
+            neo4j_uri = neo4j_host
+        else:
+            neo4j_port = vector_graph_store_config.get("port")
+            if not isinstance(neo4j_port, int):
+                raise TypeError("Neo4j port must be provided as an integer")
 
-        neo4j_uri = f"bolt://{neo4j_host}:{neo4j_port}"
+            neo4j_uri = f"bolt://{neo4j_host}:{neo4j_port}"
 
         neo4j_username = vector_graph_store_config.get("user")
         if not isinstance(neo4j_username, str):

--- a/src/memmachine/episodic_memory/session_manager/session_manager.py
+++ b/src/memmachine/episodic_memory/session_manager/session_manager.py
@@ -127,7 +127,7 @@ class SessionManager:
         sql_path = config.get("uri")
         if sql_path is None or len(sql_path) < 1:
             raise ValueError(f"""Invalid sql path: {str(config)}""")
-        if sql_path.find(":///") < 0:
+        if "postgresql" not in sql_path and sql_path.find(":///") < 0:
             sql_path = "sqlite:///" + sql_path
 
         # create empty sqlite file if it does not exist
@@ -139,6 +139,12 @@ class SessionManager:
 
         self._engine = create_engine(sql_path)
         self._session = sessionmaker(bind=self._engine)
+
+        schema = config.get("schema", "")
+        if schema:
+            for table in Base.metadata.tables.values():
+                table.schema = schema
+
         # Create all tables defined in the Base metadata if they don't exist
         Base.metadata.create_all(self._engine)
 

--- a/src/memmachine/profile_memory/storage/asyncpg_profile.py
+++ b/src/memmachine/profile_memory/storage/asyncpg_profile.py
@@ -1,3 +1,4 @@
+import functools
 import json
 import logging
 from collections.abc import Mapping
@@ -44,10 +45,6 @@ class AsyncPgProfileStorage(ProfileStorageBase):
     asyncpg implementation for ProfileStorageBase
     """
 
-    main_table = "prof"
-    junction_table = "citations"
-    history_table = "history"
-
     @staticmethod
     def build_config(config: dict[str, Any]) -> ProfileStorageBase:
         return AsyncPgProfileStorage(config)
@@ -66,32 +63,43 @@ class AsyncPgProfileStorage(ProfileStorageBase):
             raise ValueError("DB database is not in config")
         self._config = config
 
+        self.main_table = "prof"
+        self.junction_table = "citations"
+        self.history_table = "history"
+        schema = self._config.get("schema")
+        if schema is not None and schema.strip() != "":
+            schema = schema.strip()
+            self.main_table = f"{schema}.{self.main_table}"
+            self.junction_table = f"{schema}.{self.junction_table}"
+            self.history_table = f"{schema}.{self.history_table}"
+
     async def startup(self):
         """
         initializes connection pool
         """
         if self._pool is None:
+            kwargs = {}
+            # if using supabase transaction pooler, it does not support prepared statements
+            if "statement_cache_size" in self._config:
+                kwargs["statement_cache_size"] = self._config["statement_cache_size"]
             self._pool = await asyncpg.create_pool(
                 host=self._config["host"],
                 port=self._config["port"],
                 user=self._config["user"],
                 password=self._config["password"],
                 database=self._config["database"],
-                init=register_vector,
+                init=functools.partial(
+                    register_vector, schema=self._config.get("vector_schema", "public")
+                ),
+                **kwargs,
             )
 
     async def delete_all(self):
         assert self._pool is not None
         async with self._pool.acquire() as conn:
-            await conn.execute(
-                f"TRUNCATE TABLE {AsyncPgProfileStorage.main_table} CASCADE"
-            )
-            await conn.execute(
-                f"TRUNCATE TABLE {AsyncPgProfileStorage.history_table} CASCADE"
-            )
-            await conn.execute(
-                f"TRUNCATE TABLE {AsyncPgProfileStorage.junction_table} CASCADE"
-            )
+            await conn.execute(f"TRUNCATE TABLE {self.main_table} CASCADE")
+            await conn.execute(f"TRUNCATE TABLE {self.history_table} CASCADE")
+            await conn.execute(f"TRUNCATE TABLE {self.junction_table} CASCADE")
 
     async def get_profile(
         self,
@@ -103,7 +111,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
         async with self._pool.acquire() as conn:
             rows = await conn.fetch(
                 f"""
-                SELECT feature, value, tag, create_at FROM {AsyncPgProfileStorage.main_table}
+                SELECT feature, value, tag, create_at FROM {self.main_table}
                 WHERE user_id = $1
                 AND isolations @> $2
                 """,
@@ -139,8 +147,8 @@ class AsyncPgProfileStorage(ProfileStorageBase):
             result = await conn.fetch(
                 f"""
                 SELECT j.content_id
-                FROM {AsyncPgProfileStorage.main_table} p
-                LEFT JOIN {AsyncPgProfileStorage.junction_table} j ON p.id = j.profile_id
+                FROM {self.main_table} p
+                LEFT JOIN {self.junction_table} j ON p.id = j.profile_id
                 WHERE user_id = $1 AND feature = $2
                 AND value = $3 AND tag = $4
                 AND isolations @> $5
@@ -162,7 +170,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
         async with self._pool.acquire() as conn:
             await conn.execute(
                 f"""
-                    DELETE FROM {AsyncPgProfileStorage.main_table}
+                    DELETE FROM {self.main_table}
                     WHERE user_id = $1
                     AND isolations @> $2
                     """,
@@ -193,7 +201,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
             async with conn.transaction():
                 pid = await conn.fetchval(
                     f"""
-                    INSERT INTO {AsyncPgProfileStorage.main_table}
+                    INSERT INTO {self.main_table}
                     (user_id, tag, feature, value, embedding, metadata, isolations)
                     VALUES ($1, $2, $3, $4, $5, $6, $7)
                     RETURNING id
@@ -213,7 +221,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
                     return
                 await conn.executemany(
                     f"""
-                    INSERT INTO {AsyncPgProfileStorage.junction_table}
+                    INSERT INTO {self.junction_table}
                     (profile_id, content_id)
                     VALUES ($1, $2)
                 """,
@@ -236,7 +244,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
             if value is None:
                 await conn.execute(
                     f"""
-                    DELETE FROM {AsyncPgProfileStorage.main_table}
+                    DELETE FROM {self.main_table}
                     WHERE user_id = $1 AND feature = $2 AND tag = $3
                     AND isolations @> $4
                     """,
@@ -248,7 +256,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
             else:
                 await conn.execute(
                     f"""
-                    DELETE FROM {AsyncPgProfileStorage.main_table}
+                    DELETE FROM {self.main_table}
                     WHERE user_id = $1 AND feature = $2 AND tag = $3 AND value = $4
                     AND isolations @> $5
                     """,
@@ -264,7 +272,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
         async with self._pool.acquire() as conn:
             await conn.execute(
                 f"""
-            DELETE FROM {AsyncPgProfileStorage.main_table}
+            DELETE FROM {self.main_table}
             where id = $1
             """,
                 pid,
@@ -279,8 +287,8 @@ class AsyncPgProfileStorage(ProfileStorageBase):
         async with self._pool.acquire() as conn:
             stm = f"""
                 SELECT DISTINCT j.content_id, h.isolations
-                FROM {AsyncPgProfileStorage.junction_table} j
-                JOIN {AsyncPgProfileStorage.history_table} h ON j.content_id = h.id
+                FROM {self.junction_table} j
+                JOIN {self.history_table} h ON j.content_id = h.id
                 WHERE j.profile_id = ANY($1)
             """
             res = await conn.fetch(stm, pids)
@@ -308,12 +316,12 @@ class AsyncPgProfileStorage(ProfileStorageBase):
                     'value', value,
                     'metadata', JSON_BUILD_OBJECT('id', id)
                 ))
-                FROM {AsyncPgProfileStorage.main_table}
+                FROM {self.main_table}
                 WHERE user_id = $1
                 AND isolations @> $2
                 AND tag IN (
                     SELECT tag
-                    FROM {AsyncPgProfileStorage.main_table}
+                    FROM {self.main_table}
                     WHERE user_id = $1
                     AND isolations @> $2
                     GROUP BY tag
@@ -371,8 +379,8 @@ class AsyncPgProfileStorage(ProfileStorageBase):
                         , 'citations', COALESCE(
                             (
                                 SELECT JSON_AGG(h.content)
-                                FROM {AsyncPgProfileStorage.junction_table} j
-                                JOIN {AsyncPgProfileStorage.history_table} h ON j.content_id = h.id
+                                FROM {self.junction_table} j
+                                JOIN {self.history_table} h ON j.content_id = h.id
                                 WHERE p.id = j.profile_id
                             ),
                             '[]'::json
@@ -384,7 +392,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
                 + f"""
                     )
                 )
-                FROM {AsyncPgProfileStorage.main_table} p
+                FROM {self.main_table} p
                 WHERE p.user_id = $2
                 AND -(p.embedding <#> $1::vector) > $3
                 AND p.isolations @> $4
@@ -414,7 +422,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
             metadata = {}
 
         stm = f"""
-            INSERT INTO {AsyncPgProfileStorage.history_table} (user_id, content, metadata, isolations)
+            INSERT INTO {self.history_table} (user_id, content, metadata, isolations)
             VALUES($1, $2, $3, $4)
             RETURNING id, user_id, content, metadata, isolations
         """
@@ -437,7 +445,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
         isolations: dict[str, bool | int | float | str] | None = None,
     ):
         stm = f"""
-            DELETE FROM {AsyncPgProfileStorage.history_table}
+            DELETE FROM {self.history_table}
             WHERE user_id = $1 AND isolations @> $2
             AND timestamp >= {start_time} AND timestamp <= {end_time}
         """
@@ -452,7 +460,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
         is_ingested: bool = False,
     ) -> list[Mapping[str, Any]]:
         stm = f"""
-            SELECT id, user_id, content, metadata, isolations FROM {AsyncPgProfileStorage.history_table}
+            SELECT id, user_id, content, metadata, isolations FROM {self.history_table}
             WHERE user_id = $1 AND ingested = $2
             ORDER BY create_at DESC
             LIMIT $3
@@ -464,7 +472,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
 
     async def get_uningested_history_messages_count(self) -> int:
         stm = f"""
-            SELECT COUNT(*) FROM {AsyncPgProfileStorage.history_table}
+            SELECT COUNT(*) FROM {self.history_table}
             WHERE ingested=FALSE
         """
         assert self._pool is not None
@@ -477,7 +485,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
             return  # nothing to do
 
         stm = f"""
-                UPDATE {AsyncPgProfileStorage.history_table}
+                UPDATE {self.history_table}
                 SET ingested = TRUE
                 WHERE id = ANY($1::bigint[])
             """
@@ -496,7 +504,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
             isolations = {}
 
         stm = f"""
-            SELECT content FROM {AsyncPgProfileStorage.history_table}
+            SELECT content FROM {self.history_table}
             WHERE timestamp >= $1 AND timestamp <= $2 AND user_id=$3
             AND isolations @> $4
             ORDER BY timestamp ASC
@@ -519,7 +527,7 @@ class AsyncPgProfileStorage(ProfileStorageBase):
             isolations = {}
 
         query = f"""
-            DELETE FROM {AsyncPgProfileStorage.history_table}
+            DELETE FROM {self.history_table}
             WHERE user_id = $1 AND isolations @> $2 AND start_time > $3
         """
         assert self._pool is not None

